### PR TITLE
Included C++ Implementation in USACO feb20-triangles and dec19-gymnastics

### PR DESCRIPTION
**Pull request by veehz**
Included C++ Implementation in USACO feb20-triangles.mdx
Included C++ Implementation in USACO dec19-gymnastics.mdx

### CODE REVIEW

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
Add time complexity for the sols thanks!

**Review by HarryWangATX — COMMENTED** [review]
would also appreciate it if you can add a heading for each language like c++ implementation before the cppsection etc.

**Comment by veehz** [comment]
Done. C++ Headers and time complexity added.

File: solutions/USACO dec19-gymnastics.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 id: usaco-963
 source: USACO Bronze 2019 December
 title: Cow Gymnastics
-author: Krish Thawani
+author: Krish Thawani (Java), Hua Zhi Vee (C++)
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_gymnastics_bronze_dec19.html)
```
Comment (tachithanhdanh) [inline solutions/USACO dec19-gymnastics.mdx:8]:
```suggestion
[Official Analysis C++](http://www.usaco.org/current/data/sol_gymnastics_bronze_dec19.html)
```

Comment (tachithanhdanh) [reply solutions/USACO dec19-gymnastics.mdx:8]: for more clearance

Comment (HarryWangATX) [reply solutions/USACO dec19-gymnastics.mdx:8]: official analysis is standard

File: solutions/USACO dec19-gymnastics.mdx [hunk 2]
```diff
@@ -68,4 +68,43 @@ public class gymnastics {
...
+			for(int j=i+1;j<n;j++){
+				b[a[i]-1][a[j]-1] = true;
+			}
+		}
+	}
+	
+	int count = 0;
+	for(int i=0;i<n;i++){
+		for(int j=i+1;j<n;j++){
+            // If not both are true, we count it
```
Comment (tachithanhdanh) [inline solutions/USACO dec19-gymnastics.mdx:101]:
```suggestion
		// If not both are true, we count it
```
Tab indentations

File: solutions/USACO dec19-gymnastics.mdx [hunk 3]
```diff
@@ -68,4 +68,56 @@ public class gymnastics {
...
+### Solution 2
+
+We generate the ranking when we read the data.
+
+We create a 2D array of booleans which states if $A$ has a ranking higher than $B$ in any point of time.
+We read every ranking, then for all $\frac{n(n+1)}{2}$ pairs we set $b[A][B]=\text{true}$.
+
+We use $\mathcal{O}(K \cdot N^2)$ to create this array.
+
+Then we use $\mathcal{O}(N^2)$ to iterate through all pairs and check if they satisfy the condition. At least one of $b[A][B]$ and $b[B][A]$ must be true, and hence we only need to check if they are both true. If they are not both true, we increment our $\text{count}$ by 1.
```
Comment (dongliu0426) [inline solutions/USACO dec19-gymnastics.mdx:85]:
```suggestion
Then we use $\mathcal{O}(N^2)$ to iterate through all pairs and check if they satisfy the condition. At least one of $b[A][B]$ and $b[B][A]$ must be true, and hence we only need to check if they are both true. If they are not both true, we increment our $\texttt{count}$ by 1.
```

File: solutions/USACO dec19-gymnastics.mdx [hunk 4]
```diff
@@ -68,4 +68,56 @@ public class gymnastics {
...
+			for(int j=i+1;j<n;j++){
+				b[a[i]-1][a[j]-1] = true;
+			}
+		}
+	}
+	
+	int count = 0;
+	for(int i=0;i<n;i++){
+		for(int j=i+1;j<n;j++){
+		// If not both are true, we count it
```
Comment (dongliu0426) [inline solutions/USACO dec19-gymnastics.mdx:114]:
```suggestion
			// If not both are true, we count it
```

File: solutions/USACO dec19-gymnastics.mdx [hunk 5]
```diff
@@ -68,4 +68,56 @@ public class gymnastics {
 ```
 </JavaSection>
 
-</LanguageSection>
\ No newline at end of file
+<CPPSection>
+
+### Solution 1
+From [official analysis](http://www.usaco.org/current/data/sol_gymnastics_bronze_dec19.html): As the problem statement says, for each pair of cows $(A,B)$, we will count how many practice sessions cow $A$ did better than cow $B$ in. If cow $A$ did better than cow $B$ in all K practice sessions, we increment a counter, and we'll print out the value of the counter once we've looped over all pairs of cows.
```
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:74]: no need to put sol 1, just put alternate sol and ur explanatino

File: solutions/USACO dec19-gymnastics.mdx (hunk 3 above)
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:85]: Then we iterate though all pairs in n^2 time

File: solutions/USACO dec19-gymnastics.mdx [hunk 6]
```diff
@@ -68,4 +68,56 @@ public class gymnastics {
 ```
 </JavaSection>
 
-</LanguageSection>
\ No newline at end of file
+<CPPSection>
```
Comment (dongliu0426) [inline solutions/USACO dec19-gymnastics.mdx:71]: move this below `**Time complexity:** $\mathcal{O}(K \cdot N^2)$.`(i think this is breaking the build)

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-triangles.mdx [hunk 7]
```diff
@@ -59,4 +59,41 @@ public class triangles{
 ```
 
 </JavaSection>
+
+<CPPSection>
```
Comment (HarryWangATX) [inline solutions/USACO feb20-triangles.mdx:63]: kinda liked the official for this one better??? feel free to put a modified version of the official sol if u feel like its lacking in some areas

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/USACO feb20-triangles.mdx (hunk 7 above)
Comment (veehz) [reply solutions/USACO feb20-triangles.mdx:63]: Hmm you have a point, we used the same idea... I think I can remove this, how do i do it

Comment (HarryWangATX) [reply solutions/USACO feb20-triangles.mdx:63]: You can just delete ur impl and commit ig

File: solutions/USACO dec19-gymnastics.mdx [hunk 8]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
+
+### Alternate Solution
```
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:73]: use ##

File: solutions/USACO dec19-gymnastics.mdx [hunk 9]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
+
+We generate the ranking when we read the data.
+
+We create a 2D array of booleans which states if $A$ has a ranking higher than $B$ in any point of time.
+We read every ranking, then for all $\frac{n(n+1)}{2}$ pairs we set $b[A][B]=\text{true}$.
+
+We use $\mathcal{O}(K \cdot N^2)$ to create this array.
```
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:80]: remove this line

File: solutions/USACO dec19-gymnastics.mdx [hunk 10]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
+
+Then we iterate though all pairs in $\mathcal{O}(n^2)$ time and check if they satisfy the condition. At least one of $b[A][B]$ and $b[B][A]$ must be true, and hence we only need to check if they are both true. If they are not both true, we increment our $\texttt{count}$ by 1.
```
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:82]: use N^2 not n^2

Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:82]: one is false instead of not both true

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/USACO dec19-gymnastics.mdx [hunk 11]
```diff
@@ -68,4 +68,51 @@ public class gymnastics {
...
+			for(int j=i+1;j<n;j++){
+				b[a[i]-1][a[j]-1] = true;
+			}
+		}
+	}
+	
+	int count = 0;
+	for(int i=0;i<n;i++){
+		for(int j=i+1;j<n;j++){
+			// If not both are true, we count it
```
Comment (dongliu0426) [inline solutions/USACO dec19-gymnastics.mdx:109]:
```suggestion
			// If either is false, we increment the count
```

File: solutions/USACO dec19-gymnastics.mdx [hunk 12]
```diff
@@ -68,4 +68,51 @@ public class gymnastics {
 ```
 </JavaSection>
 
-</LanguageSection>
\ No newline at end of file
+<CPPSection>
+
+## Alternate Solution
+
+We generate the ranking when we read the data.
```
Comment (dongliu0426) [inline solutions/USACO dec19-gymnastics.mdx:75]:
```suggestion
We generate the ranking as we read the data.
```

File: solutions/USACO dec19-gymnastics.mdx [hunk 13]
```diff
@@ -68,4 +68,51 @@ public class gymnastics {
...
+
+We create a 2D array of booleans which states if $A$ has a ranking higher than $B$ in any point of time.
+We read every ranking, then for all $\frac{n(n+1)}{2}$ pairs we set $b[A][B]=\text{true}$.
+
+Then we iterate though all pairs in $\mathcal{O}(N^2)$ time and check if they satisfy the condition. At least one of $b[A][B]$ and $b[B][A]$ must be true, and hence we only need to check if they are both true. If one of them is false, we increment our $\texttt{count}$ by 1.
```
Comment (dongliu0426) [inline solutions/USACO dec19-gymnastics.mdx:80]: you should note that either all or exactly one must be true.

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/USACO dec19-gymnastics.mdx [hunk 14]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
-</LanguageSection>
\ No newline at end of file
+<CPPSection>
+
+## Alternate Solution
+
+We generate the ranking as we read the data.
+
+We create a 2D array of booleans which states if $A$ has a ranking higher than $B$ in any point of time.
+We read every ranking, then for all $\frac{n(n+1)}{2}$ pairs we set $b[A][B]=\text{true}$.
```
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:78]: Wrap b array with \texttt

File: solutions/USACO dec19-gymnastics.mdx [hunk 15]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
+
+Then we iterate though all pairs in $\mathcal{O}(N^2)$ time and check if they satisfy the condition. At least one of $b[A][B]$ and $b[B][A]$ must be true, and hence we only need to check if they are both true. If one of them is false, we increment our $\texttt{count}$ by 1.
```
Comment (HarryWangATX) [inline solutions/USACO dec19-gymnastics.mdx:80]: same here

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Comment by veehz** [comment]
:cake:

**Review by HarryWangATX — APPROVED** [review]

**Comment by HarryWangATX** [comment]
core member will merge

File: solutions/USACO dec19-gymnastics.mdx [hunk 16]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
-</LanguageSection>
\ No newline at end of file
+<CPPSection>
+
+## Alternate Solution
+
+We generate the ranking as we read the data.
+
+We create a 2D array of booleans which states if $A$ has a ranking higher than $B$ in any point of time.
+We read every ranking, then for all $\frac{n(n+1)}{2}$ pairs we set $\texttt{b}[A][B]=\text{true}$.
```
Comment (bqi343) [inline solutions/USACO dec19-gymnastics.mdx:78]: > $\frac{n(n+1)}{2}$

n*(n-1)/2?

File: solutions/USACO dec19-gymnastics.mdx [hunk 17]
```diff
@@ -68,4 +68,53 @@ public class gymnastics {
...
+
+Then we iterate though all pairs in $\mathcal{O}(N^2)$ time and check if they satisfy the condition. At least one of $\texttt{b}[A][B]$ and $\texttt{b}[B][A]$ must be true, and hence we only need to check if they are both true. If one of them is false, we increment our $\texttt{count}$ by 1.
```
Comment (bqi343) [inline solutions/USACO dec19-gymnastics.mdx:80]: don't use \texttt around single-letter variable names. Why is N capitalized here but not capitalized above?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/USACO dec19-gymnastics.mdx (hunk 16 above)
Comment (dongliu0426) [reply solutions/USACO dec19-gymnastics.mdx:78]: nice catch!

File: solutions/USACO dec19-gymnastics.mdx (hunk 14 above)
Comment (bqi343) [reply solutions/USACO dec19-gymnastics.mdx]: no :(

**Comment by HarryWangATX** [comment]
.

**Comment by dongliu0426** [comment]
hmmm

**Pull request merged**